gobblin-service/src/test/java/org/apache/gobblin/service/modules/orchestration/OrchestratorTest.java

```diff
@@ -81,12 +84,15 @@ public class OrchestratorTest {
 
   private FlowCatalog flowCatalog;
   private FlowSpec flowSpec;
-  private Orchestrator orchestrator;
+
+  private FlowStatusGenerator mockFlowStatusGenerator;
+  private DagManager mockDagManager;
+  private Orchestrator dagMgrNotFlowLaunchHandlerBasedOrchestrator;
   private static final String TEST_USER = "testUser";
   private static final String TEST_PASSWORD = "testPassword";
   private static final String TEST_TABLE = "quotas";
 
-  @BeforeClass
+  @BeforeMethod
```

Comment:
unfortunately lots of "testing smells" here.

first off, class-level test init doesn't play well w/ verifying mock interactions, since we want a fresh count for the exec of each test method.  relatedly it also complicates - if not outright foreclosing on - parallel test method execution.

secondly, the `@Test(dependsOnMethod = ...)` structuring is a testing anti-pattern that here obscured that `setup`-style init had been formulated instead as a test of it's own - `createTopologySpec()`.  a major clue of something wrong is that particular "@Test" method does not even exercise `Orchestrator`, our class-under-test!

for now, I've fixed this suite to use per-method setup/teardown best practices, but left it as a TODO to figure out where `createTopologySpec` actually belongs.  `createFlowSpec` and `deleteFlowSpec` also deserve attention in this same regard.

Reply:
I don't think there's anything wrong with `@BeforeClass` explicitly, it has its uses when doing expensive setup e.g. DB or Kafka setup that some other classes use. I would argue that you can have both `BeforeClass` and `BeforeMethod` depending on your needs, though for this class specifically `BeforeMethod` makes more sense for mock setup.

Reply:
I agree and now realize, after you pointed it out, that I didn't cover all test init best practice, so much as denounce what's broken about `@BeforeClass`-only init.

to encapsulate a fuller picture now, I'd advise

> prefer `@{Before,After}Method` test init to `@{Before,After}Class`".  when mocks are in use, the latter w/o the former, MOST LIKELY indicates the wrong choice, whereas having only method, but not class-level init is quite fine.  In fact the only really acceptable reason for choosing class-level init is to amortize payment of initialization tax just once per class, rather than incurring once per test case.

at the moment, I can't think of any non-performance-motivated argument to use `@BeforeClass` over just adding everything into `@BeforeMethod`

Reply:
Yeah its better to avoid state between classes when possible so beforeMethod works much better here

Reply:
interesting pt. of comparison: after the [PR check](https://github.com/apache/gobblin/actions/runs/8974993667/job/24692732545?pr=3944) test failures gave:
```
java.io.IOException: Failure creation table TestDagStateStore
```
due to:
```
java.sql.SQLNonTransientConnectionException: Too many connections
```
I realized the `ITestMetastoreDatabase` needed to be closed.

I first put `.close()` within `@AfterMethod` and the 7 `OrchestratorTest` cases took 1m49s.  I then defined an `@AfterClass` for `.close()` and the same 7 tests (all passing in both definitions) took only 33s.  hence, sticking w/ `@BeforeClass` + `@AfterClass` for that one.